nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java

```diff
@@ -177,74 +174,60 @@ private String getValue(String propertyName) {
-                converters.put(propertyType, converter);
-                return converter;
-            } catch (NoSuchMethodException | SecurityException ex) {
-                try {
-                    Method m = propertyType.getMethod("valueOf",String.class);
-                    converter = (Converter<T>) new CommonSenseConverter(m);
-                    converters.put(propertyType, converter);
-                    return converter;
-                } catch (NoSuchMethodException | SecurityException ex1) {
-                    try {
-                        Constructor c = propertyType.getConstructor(String.class);
-                        converter = (Converter<T>) new CommonSenseConverter(c);
-                        converters.put(propertyType, converter);
-                        return converter;
-                    } catch (NoSuchMethodException | SecurityException ex2) {
-                        try {
-                            Method m = propertyType.getMethod("parse", CharSequence.class);
-                            converter = (Converter<T>) new CommonSenseConverter(m);
-                            converters.put(propertyType, converter);
-                            return converter;
-                        } catch (NoSuchMethodException | SecurityException ex3) {
-                            commonSense = false;
-                        }
-                    }
-                }
-            }
+        // see if a common sense converter can be created
+        Optional<Converter<T>> commonSenseConverter = CommonSenseConverter.forClass(propertyType);
+        if (commonSenseConverter.isPresent()) {
+            converters.put(propertyType, commonSenseConverter.get());
+            return commonSenseConverter.get();
         }
-        if (converter == null) {
-            throw new IllegalArgumentException("Unable to convert value to type " + propertyType.getCanonicalName());
+
+        throw new IllegalArgumentException("Unable to convert value to type " + propertyType.getCanonicalName());
+    }
+
+
+
+    private static <T> Class<T> boxedTypeOf(Class<T> type) {
+        if (!type.isPrimitive()) {
+            return type;
+        } else if (type.equals(int.class)) {
+            return (Class<T>) Integer.class;
+        } else if (type.equals(boolean.class)) {
+            return (Class<T>) Boolean.class;
+        } else if (type.equals(long.class)) {
+            return (Class<T>) Long.class;
+        } else if (type.equals(double.class)) {
+            return (Class<T>) Double.class;
+        } else if (type.equals(float.class)) {
+            return (Class<T>) Float.class;
+        } else if (type.equals(byte.class)) {
+            return (Class<T>) Byte.class;
+        } else if (type.equals(char.class)) {
+            return (Class<T>) Character.class;
+        } else if (type.equals(short.class)) {
+            return (Class<T>) Short.class;
         }
-        return converter;
+        return type;
```

Comment:
We already established that `type` is a primitive type so should we ever get here this should mean a primitive had been added to the JVM and java that we have not covered so its more of a `throw new` situation if you ask me.

Reply:
Yes, but do we want such surprise few years later? :D But yes, initially these branches didn't list `short` and `char`.

Reply:
We could also change the name to `boxedTypeOrNot` to make the caller aware of this behaviour :D

Reply:
Actually, the only type reaching that return statement is... `void`, which is quite improbable for config property, but makes the code completely correct :D